docs/coding-guidelines/vectorization-guidelines.md

```diff
@@ -0,0 +1,1162 @@
+<255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255>
+```
+
+### Shuffle
+
+`Shuffle` creates a new vector by selecting values from an input vector using a set of indices (values that represent indexes of the input vector).
+
+```csharp
+public static Vector128<int> Shuffle(Vector128<int> vector, Vector128<int> indices)
+public static Vector128<uint> Shuffle(Vector128<uint> vector, Vector128<uint> indices)
+public static Vector128<float> Shuffle(Vector128<float> vector, Vector128<int> indices)
+public static Vector128<long> Shuffle(Vector128<long> vector, Vector128<long> indices)
+public static Vector128<ulong> Shuffle(Vector128<ulong> vector, Vector128<ulong> indices)
+public static Vector128<double> Shuffle(Vector128<double> vector, Vector128<long> indices)
+```
+
+It can be used for many things, including reversing the input:
+
+```csharp
+Vector128<int> intVector = Vector128.Create(100, 200, 300, 400);
+Console.WriteLine(intVector);
+Console.WriteLine(Vector128.Shuffle(intVector, Vector128.Create(3, 2, 1, 0)));
+```
+
+```log
+<100, 200, 300, 400>
+<400, 300, 200, 100>
+```
+
+#### Vector256.Shuffle vs Avx2.Shuffle
+
+`Vector256.Shuffle` and `Avx2.Shuffle` are not identical.
+
+`Avx2.Shuffle` is effectively `2x128-bit ops` while `Vector256.Shuffle` treats it as a "single 256-bit vector" (rather than "2x128-bit vectors"). This was done for consistency and to better map to a cross-platform mentality where `AVX-512` and `SVE` all operate on "full width".
+
+## Summary
+
+The main goal of the new `Vector128` and `Vector256` APIs is to make writing fast, vectorized code possible without becoming familiar with hardware-specific instructions and becoming an assembly language expert. Our recommendations depend on your current expertise level, software you maintain and the one you need to create:
+
+- If you are already an expert and you have vectorized your code for both `x64/x86` and `arm64/arm` code you can use the new APIs to simplify your code, but you most likely won't observe any performance gains. [#64451](https://github.com/dotnet/runtime/issues/64451) lists the places where it was/can be done in dotnet/runtime. You can use links to the merged PRs to see real-life examples.
+- If you have already vectorized your code, but only for `x64/x86` or `arm64/arm`, you can use the  new APIs to have a single, cross-platform implementation.
+- If you have already vectorized your code with `Vector<T>` you can use the new APIs to check if they can produce better code-gen.
+- If you are not familiar with hardware specific instructions or you are about to vectorize a scalar algorithm, you should start with the new `Vector128` and `Vector256` APIs. Get a solid and working implementation and eventually consider using hardware-specific methods for performance critical code paths.
+- Both managed references and unsafe pointers are dangerous to use incorrectly and each comes with their own tradeoff.
```

Comment:
> Prefer managed references over unsafe pointers to avoid pinning and safety issues.

I've removed this point from the Best practices and added following sentence to the Summary:

> Both managed references and unsafe pointers are dangerous to use incorrectly and each comes with their own tradeoff.